source/hash_table.c

```diff
@@ -722,7 +762,13 @@ void aws_hash_iter_delete(struct aws_hash_iter *iter, bool destroy_contents) {
      * underflowing to SIZE_MAX; we have to take care in aws_hash_iter_done to avoid
      * treating this as an end-of-iteration condition.
      */
+#pragma CPROVER check push
+#pragma CPROVER check disable "unsigned-overflow"
     iter->slot--;
+#pragma CPROVER check pop
+    iter->status = AWS_HASH_ITER_STATUS_DELETE_CALLED;
```

Comment:
Is this a bug fix? Please mention in the commit message if so

Reply:
see above.